### ISSUE TITLE
Memory, performance suggestion

### ISSUE THREAD

**Issue by zacharytamas** [spencercorwin/digidex-app#1]
https://github.com/spencercorwin/digidex-app/blob/579ce3ba97b6d2eed04ab45ef5fe8b90f767444a/client/src/app/app.js#L10-L16

This function expression's output is always the same. It's generally agreed that you should put these kinds outside of functions (I'd put it as a sibling of the class definition: i.e. a top-level variable in the file). As you have it now, the anonymous function returned by the expression is recreated on every call of the `render()` method and the previous (functionally identical) one is released for garbage collection. This unnecessarily thrashes memory but also affects performance because the JavaScript runtime has to evaluate that function expression, allocate memory for the new function, etc.

**Comment by spencercorwin** [comment]
Ok, yes. I think I see what you're saying. Thank you for pointing that out. This is how I refactored:
```
const App = () => (
  <Switch>
    <Route exact path="/" component={Login} />
    <Route path="/profile" component={Home} />
    <Route component={NoMatch} />
  </Switch>
);

export default App;
```

**Comment by zacharytamas** [comment]
Yeah that is more succinct and is only declared once. It also doesn't result in a doubly-nested `<Switch>` like before. I didn't notice that the first time I looked at the code, actually. 

Also, I'm sure it wasn't intentional, but you defined `App` as a class in the module's scope and then you declared another `App` constant inside the `render()` function's scope. This is called "shadowing". In a small file like this it's fairly easy to see what's going on but in a larger one it could be confusing if you have high-level scope variables like that that can suddenly point to different values in other places in the file. You can reuse variable names, of course, but generally reusing one that is also declared in a higher scope is a confusing practice.

Do you use a JavaScript linter such as [ESLint](https://eslint.org/)? [Here](https://eslint.org/docs/rules/no-shadow)'s the ESLint rule that would warn you about this specific thing. 

It is well worth your time to get familiar with ESLint and the basics of how to set it up: almost any professional software project you join in the future will be using tools like this to keep consistency between developer styles on their teams.

**Comment by spencercorwin** [comment]
Yeah I did not mean to define `App` twice. There’s definitely a lot of low-hanging fruit for refactoring when I make a second pass.

I have not used linters in any of my own projects yet. But I have recently been exposed to a linter for a few open source PRs I submitted. My local commits went through a linter before allowing to commit. I can definitely see the value. Thank you for the suggestion.
